drivers/mtd_sdmmc/mtd_sdmmc.c

```diff
@@ -0,0 +1,254 @@
+        }
+        size = MIN(size, SDMMC_SDHC_BLOCK_SIZE - offset);
+        DEBUG("mtd_sdmmc_read_page: read %" PRIu32 " bytes at offset %" PRIu32 "\n",
+              size, offset);
+        memcpy(buff, (uint8_t *)dev->work_area + offset, size);
+        return size;
+#else
+        return -ENOTSUP;
+#endif
+    }
+
+    int err = sdmmc_read_blocks(mtd_sd->sdmmc, page, SDMMC_SDHC_BLOCK_SIZE,
+                                size / SDMMC_SDHC_BLOCK_SIZE, buff, NULL);
+    if (err) {
+        DEBUG("mtd_sdmmc_read_page: error %d\n", err);
+        return -EIO;
+    }
+    return size;
+}
+
+static int mtd_sdmmc_write_page(mtd_dev_t *dev, const void *buff, uint32_t page,
+                                 uint32_t offset, uint32_t size)
+{
+    mtd_sdmmc_t *mtd_sd = (mtd_sdmmc_t*)dev;
+
+    DEBUG("mtd_sdmmc_write_page: page:%" PRIu32 " offset:%" PRIu32 " size:%" PRIu32 "\n",
+          page, offset, size);
+
+    if (offset || size % SDMMC_SDHC_BLOCK_SIZE) {
+#if IS_USED(MODULE_MTD_WRITE_PAGE)
+        if (dev->work_area == NULL) {
+            DEBUG("mtd_sdmmc_write_page: no work area\n");
+            return -ENOTSUP;
+        }
+
+        if (sdmmc_read_blocks(mtd_sd->sdmmc, page, SDMMC_SDHC_BLOCK_SIZE,
+                              1, dev->work_area, NULL)) {
+            return -EIO;
+        }
+
+        size = MIN(size, SDMMC_SDHC_BLOCK_SIZE - offset);
+        DEBUG("mtd_sdmmc_write_page: write %" PRIu32 " bytes at offset %" PRIu32 "\n",
+              size, offset);
+        memcpy((uint8_t *)dev->work_area + offset, buff, size);
+        if (sdmmc_write_blocks(mtd_sd->sdmmc, page, SDMMC_SDHC_BLOCK_SIZE,
+                               1, dev->work_area, NULL)) {
+            return -EIO;
+        }
+#else
+        return -ENOTSUP;
+#endif
+    }
+    else {
+        int err = sdmmc_write_blocks(mtd_sd->sdmmc, page, SDMMC_SDHC_BLOCK_SIZE,
+                                     size / SDMMC_SDHC_BLOCK_SIZE, buff, NULL);
+        if (err) {
+            DEBUG("mtd_sdmmc_write_page: error %d\n", err);
+            return -EIO;
+        }
+    }
+    return size;
+}
+
+static int mtd_sdmmc_erase_sector(mtd_dev_t *dev, uint32_t sector, uint32_t count)
+{
+#if IS_ACTIVE(CONFIG_MTD_SDMMC_ERASE) && IS_USED(MODULE_MTD_WRITE_PAGE)
+    mtd_sdmmc_t *mtd_sd = (mtd_sdmmc_t*)dev;
+
+    DEBUG("mtd_sdmmc_erase_sector: sector: %" PRIu32 " count: %" PRIu32 "\n",
+          sector, count);
+
+    if (dev->work_area == NULL) {
+        DEBUG("mtd_sdmmc_erase_sector: no work area\n");
+        return -ENOTSUP;
+    }
+    memset(dev->work_area, 0, SDMMC_SDHC_BLOCK_SIZE);
+    while (count) {
+        if (sdmmc_write_blocks(mtd_sd->sdmmc, sector, SDMMC_SDHC_BLOCK_SIZE,
+                               1, dev->work_area, NULL)) {
+            return -EIO;
+        }
+        --count;
+        ++sector;
+    }
+#else
+    (void)dev;
+    (void)sector;
+    (void)count;
+    mtd_sdmmc_t *mtd_sd = (mtd_sdmmc_t*)dev;
+    if (IS_ACTIVE(CONFIG_MTD_SDMMC_ERASE)) {
```

Comment:
Why is this conditional?
Always just calling `sdmmc_erase_blocks()` works, we don't even need the special case for when `mtd_write_page` is implemented.

Reply:
See comment https://github.com/RIOT-OS/RIOT/pull/20180#issuecomment-1854451957